### ISSUE TITLE
Bug: Video processed by fl_time has no duration

### ISSUE THREAD

**Issue by LaoYuanPython** [Zulko/moviepy#1209]
My Code is:

```
from  moviepy.editor import *
from  moviepy.Clip import *
clip = VideoFileClip(r"F:\video\WinBasedWorkHard.mp4")	
newclip = clip.fl_time(lambda t: 2*t, apply_to=['mask', 'audio'],keep_duration=False)
newclip.write_videofile(r"F:\video\WinBasedWorkHard_new.mp4")
```

#### Expected Behavior

newclip was saved to file successful
#### Actual Behavior

```
Traceback (most recent call last):
  File "<pyshell#5>", line 1, in <module>
    newclip.write_videofile(r"F:\video\WinBasedWorkHard_new.mp4")
  File "<decorator-gen-55>", line 2, in write_videofile
  File "C:\Program Files\Python37\lib\site-packages\moviepy\decorators.py", line 52, in requires_duration
    raise ValueError("Attribute 'duration' not set")
ValueError: Attribute 'duration' not set
```

#### Steps to Reproduce the Problem

1、unzip the file WinBasedWorkHard.zip by winrar or other tools
2、excute the scripts in IDLE (Python 3.7.2 64bit)
3、show newclip and clip's duration

Note:
1.if any question about this issues,please send mail to :jwp139@139.com
2.Some examples in Clip.py has errors:
`.fx.`:
1>
`clip.fx( volumex, 0.5).fx( resize, 0.3).fx( mirrorx )` Is equivalent `resize( volumex( mirrorx( clip ), 0.5), 0.3)` should be `mirrorx(resize (volumex ( clip , 0.5), 0.3))`
`.fl_time`:
2>
`newclip = clip.fl_time(lambda: 2*t, apply_to=['mask', 'audio'])` should be `newclip = clip.fl_time(lambda t: 2*t, apply_to=['mask', 'audio'])`
3>
`newclip = clip.fl_time(lambda: 3-t)` should be `newclip = clip.fl_time(lambda t: 3-t)`

[WinBasedWorkHard.zip](https://github.com/Zulko/moviepy/files/4707008/WinBasedWorkHard.zip)

#### Specifications

  - Python Version:Python 3.7.2 64bit
  - Moviepy Version:1.03
  - Platform Name:win7 
  - Platform Version:win7 64bit

**Comment by tburrows13** [comment]
Hello, I don't think that your problem is a bug with MoviePy. You have called `fl_time` with `keep_duration=False`, so the resultant clip is not supposed to have a `duration` attribute. You can set it yourself with, for example, `newclip = newclip.set_duration(clip.duration // 2)`, or use `keep_duration=True` to allow it to have the same duration as `clip`.

Thanks for noting the mistakes in the documentation. 2) and 3) have been noted in #1129 and fixed in #1076, but I haven't seen 1) before, so I will be sure to fix that, thank you!

**Comment by LaoYuanPython** [comment]
Zulko/moviepy：您好！
 I see,thanks very much!
    此致
敬礼！
 
                                                  姜文平
                                                    2020-06-01
----------------------------------------------------------------------
如有问题，请随时联系！联系方式：
姜文平
业务支撑中心
中国移动集团湖北有限公司
湖北省武汉市东西湖区金银湖路2号
邮政编码：430040
电话 13971275598 
电子邮件：jwp139@139.com
-------------------------------------------------------------------------
 
From: Tom Burrows
Date: 2020-06-01 04:38
To: Zulko/moviepy
CC: LaoYuanPython; Author
Subject: Re: [Zulko/moviepy] Bug: Video processed by fl_time has no duration (#1209)
Hello, I don't think that your problem is a bug with MoviePy. You have called fl_time with keep_duration=False, so the resultant clip is not supposed to have a duration attribute. You can set it yourself with, for example, newclip = newclip.set_duration(clip.duration // 2), or use keep_duration=True to allow it to have the same duration as clip.
Thanks for noting the mistakes in the documentation. 2) and 3) have been noted in #1129 and fixed in #1076, but I haven't seen 1) before, so I will be sure to fix that, thank you!

**Comment by tburrows13** [comment]
I think that this can be closed then. I've fixed the Clip.fx docstring in #1211.
